parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,331 @@
+   * @return A caller-managed BlockPropertyItem.
+   */
+  static ParseBlockPropertyItem* GetCompressAttributesItem(int line_number,
+                                                           int column_number,
+                                                           PtrList<ParseString> *attributes) {
+    ParseBlockPropertyItem* item = new ParseBlockPropertyItem(line_number,
+                                                              column_number,
+                                                              Property::kCompress);
+    item->values_.reset(attributes);
+    return item;
+  }
+
+  /**
+   * @brief Useful in the case where the parser finds a sort attribute.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @param attribute ParseString which contains a column name to sort.
+   * @return A caller-managed BlockPropertyItem.
+   */
+  static ParseBlockPropertyItem* GetSortItem(int line_number,
+                                             int column_number,
+                                             ParseString *attribute) {
+    ParseBlockPropertyItem* item = new ParseBlockPropertyItem(line_number,
+                                                              column_number,
+                                                              Property::kSort);
+    item->values_->push_back(attribute);
+    return item;
+  }
+
+  /**
+   * @brief Useful in the case where the parser finds a block type.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @param attribute ParseString which contains a column name to sort.
+   * @return A caller-managed BlockPropertyItem.
+   */
+  static ParseBlockPropertyItem* GetTypeItem(int line_number,
+                                             int column_number,
+                                             ParseString *type_string) {
+    ParseBlockPropertyItem* item = new ParseBlockPropertyItem(line_number,
+                                                              column_number,
+                                                              Property::kType);
+    item->values_->push_back(type_string);
+    return item;
+  }
+
+  /**
+   * @return Name of the parser construct represented by this class.
+   */
+  std::string getName() const override {
+    return std::string("BlockPropertyItem");
+  }
+
+  /**
+   * @brief The type of property being described.
+   *
+   * @return The specific block property which this instance is describing.
+   */
+  Property property() const {
+    return property_;
+  }
+
+  /**
+   * @return A list of ParseString values which the parser has captured to
+   *         describe this property. This may be empty in the case of 'COMPRESS ALL'.
```

Comment:
This comment is no longer accurate. On the one hand, it may be empty in the case of a numeric block size property. On the other hand, an "ALL" string is now added to the values when `GetCompressAllItem()` is called.